packages/contracts/contracts/LenderCommitmentForwarder.sol

```diff
@@ -75,113 +129,206 @@ contract LenderCommitmentForwarder is TellerV2MarketForwarder {
         TellerV2MarketForwarder(_protocolAddress, _marketRegistry)
     {}
 
+  /**
+     * @notice Created a loan commitment from a lender to a market.
+     * @param _marketId The marketId in which the commitment is valid
+     * @param _principalTokenAddress The address of the asset being committed.
+     * @param _maxPrincipal Amount of tokens being committed by the lender.
+
+     * @param _collateralTokenAddress The address of the collateral asset required for the loan.
+     * @param _maxPrincipalPerCollateralAmount Amount of loan principal allowed per each collateral amount expressed in raw value regardles of token decimals.
+     * @param _collateralTokenType The token type of the collateral 
+
+     * @param _maxLoanDuration Length of time, in seconds that the lender's capital can be lent out for.
+     * @param _minInterestRate Minimum Annual percentage to be applied for loans using the lender's capital.
+     * @param _expiration Expiration time in seconds, when the commitment expires.
+     * @param _borrower The address of the borrower who is allowed to accept this commitment.  Use the zero address to wildcard for any borrower.
+     */
+    function createCommitment(
+        uint256 _marketId,
+        address _principalTokenAddress,
+        uint256 _maxPrincipal,
+        address _collateralTokenAddress,
+        uint256 _collateralTokenId,
+        uint256 _maxPrincipalPerCollateralAmount,
+        CommitmentCollateralType _collateralTokenType,
+        uint32 _maxLoanDuration,
+        uint16 _minInterestRate,
+        uint32 _expiration,
+        address _borrower
+    ) public returns (uint256 commitmentId) {
+        commitmentId = commitmentCount++;
+
+        lenderMarketCommitments[commitmentId] = Commitment({
+            maxPrincipal: _maxPrincipal,
+            expiration: _expiration,
+            maxDuration: _maxLoanDuration,
+            minInterestRate: _minInterestRate,
+            collateralTokenAddress: _collateralTokenAddress,
+            collateralTokenId: _collateralTokenId,
+            maxPrincipalPerCollateralAmount: _maxPrincipalPerCollateralAmount,
+            collateralTokenType: _collateralTokenType,
+            lender: _msgSender(),
+            marketId: _marketId,
+            principalTokenAddress: _principalTokenAddress,
+            borrower: _borrower
+        });
```

Comment:
tbh don't really like how similar `createCommitment` and `updateCommitment` are (minus the necessary lender check)

It might just make sense instead of defining each argument and then constructing the struct, we make the struct the argument and just set it to the mapping - 1 liner but might need to take the `lender` out and maybe just have it in a separate mapping?

```suggestion
    function createCommitment(
    		Commitment calldata _commitment
    ) public returns (uint256 commitmentId) {
        commitmentId = commitmentCount++;

        commitments[commitmentId] = _commitment;
        commitmentLender[commitmentId] = _msgSender();
```

Reply:
done